backend/grant/task/jobs.py

```diff
@@ -86,8 +86,48 @@ def process_task(task):
                     'account_settings_url': make_url('/profile/settings?tab=account')
                 })
 
+class ContributionExpired:
+    JOB_TYPE = 3
+
+    def __init__(self, contribution):
+        self.contribution = contribution
+    
+    def blobify(self):
+        return {
+            "contribution_id": self.contribution.id,
+        }
+    
+    def make_task(self):
+        from .models import Task
+        task = Task(
+            job_type=self.JOB_TYPE,
+            blob=self.blobify(),
+            execute_after=self.contribution.date_created + timedelta(hours=24),
+        )
+        db.session.add(task)
+        db.session.commit()
+
+    @staticmethod
+    def process_task(task):
+        from grant.proposal.models import ProposalContribution
```

Comment:
avoiding circular deps?

Reply:
Yep, unfortunately all of the job classes seem to need to handle it this way.